File: 2-DNSAttackCSharp/SSIN/UDPSocket.cs
using System;
using System.Net;
using System.Net.Sockets;
using System.Text;
using System.Threading;

namespace DNS_Attack {
    public class UDPSocket {
        private IPEndPoint localIpEndPoint;
        private UdpClient udpClient;

        public UDPSocket(int port) {
            localIpEndPoint = new IPEndPoint(IPAddress.Any, port);
            udpClient = new UdpClient(localIpEndPoint);
        }

        public void SendUDPPacket(string host, int port, byte[] data) {
            IPAddress[] addresses = Dns.GetHostAddresses(host);
            if (addresses.Length == 0)
                throw new Exception("No IP found for the indicated host.");
            udpClient.Send(data, data.Length, new IPEndPoint(IPAddress.Parse(addresses[0].ToString()), port));
        }

        public UDPPacketData ReceiveUDPPacket() {
            IPEndPoint remoteIpEndPoint = new IPEndPoint(IPAddress.Any, 0);
            byte[] data = udpClient.Receive(ref remoteIpEndPoint);
            return new UDPPacketData(localIpEndPoint, remoteIpEndPoint, data);
        }

        public void Close() {
            udpClient.Close();
        }
    }
}

File: 1-DNSAttackJava/UDPSocket.java
import java.io.IOException;
import java.net.DatagramPacket;
import java.net.DatagramSocket;
import java.net.InetAddress;
import java.net.SocketException;

public class UDPSocket {
    private final DatagramSocket datagramSocket;
    private static final int arrayLength = 1024 * 4;

    public UDPSocket(int port) throws SocketException {
        this.datagramSocket = new DatagramSocket(port);
    }

    public void sendUDPPacket(String host, int port, byte[] data) throws IOException {
        InetAddress address = InetAddress.getByName(host);
        DatagramPacket packet = new DatagramPacket(data, data.length, address, port);
        datagramSocket.send(packet);
    }

    public DatagramPacket receiveUDPPacket() throws IOException {
        byte[] data = new byte[arrayLength];
        DatagramPacket packet = new DatagramPacket(data, data.length);
        datagramSocket.receive(packet);
        return packet;
    }

    public void close() {
        datagramSocket.close();
    }
}
